File: crates/asteroid-miner-world/src/pawn.rs
use asteroid_miner_types::*;
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

use crate::SkillState;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Pawn {
    pub id: PawnId,
    pub first_name: String,
    pub last_name: String,
    pub nickname: Option<String>,
    pub age: u16,
    pub sex: Sex,

    pub origin: OriginId,
    pub career: CareerId,

    pub traits: Vec<TraitId>,
    pub skills: HashMap<SkillId, SkillState>,
}

File: packages/model/src/runtime/pawn.ts
export type SkillState = {
  level: number;
  xp: number;
};

export type Pawn = {
  id: string;
  faction: string;
  firstName: string;
  middleName: string;
  lastName: string;
  nickname: string | null;
  sex: 'Male' | 'Female';
  age: number;
  origin: string;
  career: string;
  skills: Record<string, SkillState>;
  traits: string[];
};
